src/mlpack/methods/dbscan/random_point_selection.hpp

```diff
@@ -45,14 +53,20 @@ class RandomPointSelection
         ++found;
 
       if (found > index)
+      {
+        unvisited[i].flip(); // Set unvisited point to visited point
         return i;
+      }
     }
-
     return 0; // Not sure if it is possible to get here.
   }
-};
 
+ private:
+  // Bitmask for unvisited points, If true, mean unvisited.
+  boost::dynamic_bitset<> unvisited;
```

Comment:
I don't remember why I used `dynamic_bitset<>`, but you can probably change this with `std::vector<bool>` to avoid all the overhead of including boost parts (that can actually be a significant compiler slowdown since Boost is so large).

Reply:
Yes, I will do it!!